src/storage-rbox/rbox-save.cpp

```diff
@@ -712,9 +704,9 @@ void rbox_transaction_save_commit_post(struct mail_save_context *_ctx,
   mail_index_sync_set_commit_result(r_ctx->sync_ctx->index_sync_ctx, result);
 
   if (rbox_sync_finish(&r_ctx->sync_ctx, TRUE) < 0) {
-    r_ctx->failed = TRUE;
+    r_ctx->failed = r_ctx->failed ? TRUE : FALSE;
```

Comment:
Is this correct? The same field on both sides of the expression?

Reply:
yes, it is not necessary anymore to preserve the failed state here. in case rbox_sync_finish fails the r_ctx->failed state is true.